### PR TITLE
[2021.1][Yamato] Misc backports

### PR DESCRIPTION
**Pull request by sophiaaar**
### Purpose of this PR
* backport #3034 (allow multiple performance projects)
* backport #3168 (add 2d performance testing)
* backport #3215 (xr cache server jobs)
* backport #3107 
* backport #3129 

---
### Testing status
https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/2021.1%252Fperformance%252F2d-testing

---
### Comments to reviewers
* This PR also adds some jobs back that were missing from the nightly
* Removed Lost Crypt win dx12 playmode editor from ci due to case 1308059

### CODE REVIEW

File: .yamato/all-urp_performance_lostcrypt.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      -  path: .yamato/urp_performance_lostcrypt-iphone-metal.yml#URP_Performance_LostCrypt_iPhone_Metal_performance_playmode_runtime_il2cpp_Linear_2021.1
+         rerun: on-new-revision
+      -  path: .yamato/urp_performance_lostcrypt-iphone-metal.yml#URP_Performance_LostCrypt_iPhone_Metal_performance_build_editmode_il2cpp_Linear_2021.1
+         rerun: on-new-revision
+    variables:
+        UTR_VERSION: "current"
+        TEST_FILTER: .*
+        LOST_CRYPT_BRANCH: 21.2/urp-perf-test
+        LOST_CRYPT_REVISION: eb3405aada97fdaf2b033e4f7bc7e47fb4589c3a
+Nightly_URP_Performance_LostCrypt_2021.1:
```
Comment (theo-pnv) [inline .yamato/all-urp_performance_lostcrypt.yml:44]: Isn't it missing from the parent nightly? I can't see where this one is triggered.

File: .yamato/config/_abv.metafile [hunk 2]
```diff
@@ -12,6 +12,16 @@ abv: # specify the editor versions for which to create the expression trigger fo
     - il2cpp
   color_spaces: 
     - Linear
+nightly:
+  extra_dependencies: # additional project jobs to run nightly
+    - project: URP_Performance_BoatAttack
```
Comment (theo-pnv) [inline .yamato/config/_abv.metafile:17]: I can't find where these pipelines were _actually_ added in the nightly parent in _abv.yml

**Review by theo-pnv — APPROVED** [review]

File: .yamato/all-urp_performance_lostcrypt.yml (hunk 1 above)
Comment (sophiaaar) [reply .yamato/all-urp_performance_lostcrypt.yml:44]: fixed

File: .yamato/config/_abv.metafile (hunk 2 above)
Comment (sophiaaar) [reply .yamato/config/_abv.metafile:17]: fixed

**Pull request merged**